src/math/WadRayMath.sol

```diff
@@ -116,4 +116,70 @@ library WadRayMath {
             }
         }
     }
+
+    /// @notice Computes the wad-based weighted average (x * (1 - p) + y * p), rounded half up.
+    /// @param x The first value, with a weight of 1 - weight.
+    /// @param y The second value, with a weight of weight.
+    /// @param weight The weight of y, and complement of the weight of x (in wad).
+    /// @return z The result of the wad-based weighted average.
+    function wadWeightedAvg(
+        uint256 x,
+        uint256 y,
+        uint256 weight
+    ) internal pure returns (uint256 z) {
+        // Must revert if
```

Comment:
```suggestion
        // Underflow/overflow if
```
In theory there is no reason to revert other than `weight > WAD`, but here we also revert because of the way we do the computation.